sql/catalyst/src/test/scala/org/apache/spark/sql/connector/InMemoryTable.scala

```diff
@@ -274,11 +274,13 @@ class InMemoryTable(
         this
       }
 
-      override def buildForBatch(): BatchWrite = writer
```

Comment:
If I understand correctly, this change is optional (shouldn't break anything even this is not change), as we're trying to guarantee backward compatibility, right? I think it's better to change this - just wanted to check about compatibility.

Reply:
Since this is a test module, it's okay to change.

Reply:
Yeah, we could leave it as is. I changed because it would previously fail if a data source implemented the build method only. 

In the next PR, I am going to implement RequireaDistributionAndOrdering.

Reply:
Sorry for the confusion. I didn't mean we should leave it as it is. I just meant to confirm whether the test won't fail even without the fix, to make sure we guarantee compatibility.

Reply:
Yep, I confirm it is compatible, @HeartSaVioR.